### PR TITLE
[Fix:#722] : Check errors for LoadInstances() call 

### PR DESCRIPTION
**Pull request by haardikdharma10**
#### Summary

This PR aims to solve #722 

#### Ticket Link

#722

### CODE REVIEW

**Review by jfrerich — APPROVED** [review]
Thanks, @haardikdharma10.  This was actually a bug on my part.

**Comment by haardikdharma10** [comment]
> Thanks, @haardikdharma10. This was actually a bug on my part.

No worries:)

**Review by mickmister — CHANGES_REQUESTED** [review]
Thanks for tackling this task @haardikdharma10 :+1: Looking through the codebase, it looks like there is some error checks missing in `instances.go`. Do you mind checking out that file to find the other places to check for this error?

**Comment by haardikdharma10** [comment]
@mickmister, sure thing.

**Comment by haardikdharma10** [comment]
Hi @mickmister, I face some difficulties to check for errors in the instances.go file. Do you mind creating a separate ticket for that and maybe we can merge this PR as it solves #722 ?

**Comment by mickmister** [comment]
> Hi @mickmister, I face some difficulties to check for errors in the instances.go file

@haardikdharma10 Do you mind explaining the difficulties you've encountered?

> Do you mind creating a separate ticket for that and maybe we can merge this PR as it solves #722 ?

I would think that the cases in `instances.go` are part of the same task. Is there a reason this is not the case? We can change the scope of the task if necessary.

**Comment by jfrerich** [comment]
Hi @haardikdharma10, have you had a chance to review the response from @mickmister?

**Comment by haardikdharma10** [comment]
Hi @jfrerich, sorry for the delay on this. Just a quick question, what should be the `http.Request` error here, when the instances fail to load? Will it be `http.StatusBadRequest` or something else?

**Comment by mickmister** [comment]
> what should be the `http.Request` error here, when the instances fail to load? Will it be `http.StatusBadRequest` or something else?

@haardikdharma10 I think `http.StatusInternalServerError` should be used because the issue is purely on the server, such as a database error. `http.StatusBadRequest` should be used when the client's request has a missing value for example.

**Comment by haardikdharma10** [comment]
Got it! Thank you @mickmister, and sorry if I ask a lot of questions :D

File: server/instances.go [hunk 1]
```diff
@@ -364,8 +366,11 @@ func (p *Plugin) httpAutocompleteUserInstance(w http.ResponseWriter, r *http.Req
 			Item: info.User.DefaultInstanceID.String(),
 		})
 	}
-
 	instances, err := p.instanceStore.LoadInstances()
+	if err != nil {
+		return respondErr(w, http.StatusInternalServerError, err)
```
Comment (mickmister) [inline server/instances.go:371]:
```suggestion
		return respondErr(w, http.StatusInternalServerError, errors.Wrap(err, "failed to load instances"))
```

File: server/instances.go [hunk 2]
```diff
@@ -401,6 +406,10 @@ func (p *Plugin) httpAutocompleteInstalledInstanceWithAlias(w http.ResponseWrite
 	}
 
 	instances, err := p.instanceStore.LoadInstances()
+	if err != nil {
+		return respondErr(w, http.StatusInternalServerError, err)
```
Comment (mickmister) [inline server/instances.go:410]:
```suggestion
		return respondErr(w, http.StatusInternalServerError, errors.Wrap(err, "failed to load instances"))
```

File: server/instances.go [hunk 3]
```diff
@@ -296,6 +296,9 @@ func (p *Plugin) resolveUserInstanceURL(user *User, instanceURL string) (types.I
 	}
 
 	instances, err := p.instanceStore.LoadInstances()
+	if err != nil {
+		return "", err
```
Comment (mickmister) [inline server/instances.go:300]:
```suggestion
		return "", errors.Wrap(err, "failed to load instances")
```

**Review by mickmister — CHANGES_REQUESTED** [review]
Thanks @haardikdharma10! I have one last request here to include some context on the errors. Otherwise looks great :+1:

**Comment by haardikdharma10** [comment]
Applied your suggestions. Thank you for your time @mickmister :)

**Review by mickmister — APPROVED** [review]
LGTM thanks @haardikdharma10!!

**Pull request merged**